website/blog/2023-11-11-AgentEval/index.mdx

```diff
@@ -0,0 +1,144 @@
+   * _The optimal solution exits_ -  there are tasks where only one solution is possible. For example, if you ask your assistant to turn on the light, the success of this task is clearly defined, and there is only one way to accomplish it.
+   * _Multiple solutions exist_ - "Increasingly, we observe situations where multiple trajectories of agent behavior can lead to either success or failure. In such cases, it is crucial to differentiate between the various successful and unsuccessful trajectories. For example, when you ask the agent to suggest you a food recipe or tell you a joke.
+
+In our AgentEval framework, we are currently focusing on  _Executable Tasks_.
+
+## AgentEval Framework
+
+Our previous research on [assistive agents in Minecraft](https://github.com/microsoft/iglu-datasets) suggested that the most optimal way to obtain human judgments is to present humans with two agents side by side and ask for preferences. Interestingly, according to our earlier work, humans can develop criteria to explain why they prefer one agent over another in the setup of pair-wise comparison. For example, 'the first agent was faster in execution,' or 'the second agent moves more naturally.' So, the comparative nature led humans to come up with a list of criteria that helps to infer the utility of the task. With this idea in mind, we designed AgentEval (shown in Fig. 1), where we employ LLMs to help us understand, verify, and assess the task *utility* for the multi-agent system. Namely:
+
+* The goal of `CriticAgent` is to suggest the list of criteria $(c_1,\dots, c_n)$, that can be used to assess task utility. This is an example of how `CriticAgent` is defined using `Autogen`:
+
+
+```python
+critic = autogen.AssistantAgent(
+    name="critic",
+    llm_config={"config_list": config_list},
+    system_message="""You are a helpful assistant. You suggest criteria for evaluating different tasks. They should be distinguishable, quantifiable, and not redundant.
+    Convert the evaluation criteria into a dictionary where the keys are the criteria.
+    The value of each key is a dictionary as follows {"description": criteria description, "accepted_values": possible accepted inputs for this key}
+    Make sure the keys are criteria for assessing the given task. "accepted_values" include the acceptable inputs for each key that are fine-grained and preferably multi-graded levels. "description" includes the criterion description.
+    Return only the dictionary."""
+)
+```
+
+* The goal of `QuantifierAgent` is to quantify each of the suggested criteria in the following way: $(c_1=a_1, \dots, c_n=a_n)$, providing us with an idea of the utility of this system for the given task -- $U(t_i)$. Here is an example of how it can be defined:
+

```

Comment:
It may be useful for the reader here to know whether the critic or the quantifier agent requires as an input the definition of the task itself or few shot examples from the dataset. Or whether it is useful at all to do so (even if not required).